backend/instance.ts

```diff
@@ -80,6 +81,17 @@ export class Instances {
     });
   }
 
+  delete(id: number) {
+    let instance = this.instances.get(id)!;
```

Comment:
You should really handle the undefined case explicitly instead of working around it with `!`. If the instance doesn't exist and that really shouldn't happen, throw an `Error` object so you get some debugging info.